intelmq/bin/intelmqsetup.py

```diff
@@ -17,59 +22,135 @@
 import glob
 import os
 import shutil
+import stat
 import sys
 import pkg_resources
 
-from pwd import getpwuid
+from grp import getgrnam
+from pwd import getpwuid, getpwnam
+from typing import Optional
+
+try:
+    import intelmq_api
+except ImportError:
+    intelmq_api = None
 
+from termstyle import red
 from intelmq import (CONFIG_DIR, DEFAULT_LOGGING_PATH, ROOT_DIR, VAR_RUN_PATH,
                      VAR_STATE_PATH, BOTS_FILE, STATE_FILE_PATH)
 from intelmq.bin.intelmqctl import IntelMQController
 
 
-def intelmqsetup(ownership=True, state_file=STATE_FILE_PATH):
-    if os.geteuid() != 0 and ownership:
-        sys.exit('You need to run this program as root (for setting file ownership)')
+MANAGER_CONFIG_DIR = os.path.join(CONFIG_DIR, 'manager/')
+FILE_OUTPUT_PATH = os.path.join(VAR_STATE_PATH, 'file-output/')
 
+
+def basic_checks(skip_ownership):
+    if os.geteuid() != 0 and not skip_ownership:
+        sys.exit(red('You need to run this program as root for setting file ownership!'))
     if not ROOT_DIR:
-        sys.exit('Not a pip-installation of IntelMQ, nothing to initialize.')
-
-    create_dirs = ('%s/file-output' % VAR_STATE_PATH,
-                   VAR_RUN_PATH,
-                   DEFAULT_LOGGING_PATH,
-                   CONFIG_DIR)
-    for create_dir in create_dirs:
-        if not os.path.isdir(create_dir):
-            os.makedirs(create_dir, mode=0o755,
-                        exist_ok=True)
-            print('Created directory %r.' % create_dir)
+        sys.exit(red('Not a pip-installation of IntelMQ, nothing to initialize.'))
+
+    if skip_ownership:
+        return
+    try:
+        getpwnam('intelmq')
+    except KeyError:
+        sys.exit(red("User 'intelmq' does not exist. Please create it and then re-run this program."))
+    try:
+        getgrnam('intelmq')
+    except KeyError:
+        sys.exit(red("Group 'intelmq' does not exist. Please create it and then re-run this program."))
+
+
+def create_directory(directory: str, octal_mode: int, readable_mode: str):
+    if not os.path.isdir(directory):
+        os.makedirs(directory, mode=octal_mode,
+                    exist_ok=True)
+        print(f'Created directory {directory!r} with permissions {readable_mode}.')
+    else:
+        actual_mode = stat.filemode(os.stat(directory).st_mode)
```

Comment:
why compare the string representation if you can compare the actual mode?  (`actual_mode` should probably be `current_mode`)